applications/StructuralMechanicsApplication/custom_response_functions/README.md

```diff
@@ -0,0 +1,248 @@
+    * Point load (wraps the PointLoadCondition)
+
+- Structural adjoint elements:
+    * One-dimensional elements:
+       	* Linear 3D beam element (wraps the CrBeamElementLinear3D2N)
+    * Two-dimensional elements:
+        * Thin triangular shell (wraps the ShellThinElement3D3N)
+
+*Please note:*
+The adjoint elements and conditions wrap elements/conditions of the Structural Mechanics Application and can call its public functions.  The main task of the adjoint elements/conditions is to derive different quantities (e.g. the right hand side or post-processing results like stresses) with respect to the design variable or state.
+
+### Usage:
+In order to perform a sensitivity analysis for one response function, the solutions of two linear static problems are necessary: The primal and the adjoint problem.
+
+*Please note:*
+For the solution of the two problems different kind of variables are used in order to store the results. For the primal problem the usual variables ```DISPLACEMENT``` and ```ROTATION``` and for the adjoint problem ```ADJOINT_DISPLACEMENT``` and ```ADJOINT_ROTATION``` are used.
+
+#### Definition of the Primal Problem
+The primal problem can be defined by the regular input files which are needed for an usual linear static analysis. As only difference the output process of the HDF5Application has to be added to the ```list_other_processes``` in the project parameters:
+
+```python
+    "list_other_processes" :[{
+        "kratos_module" : "KratosMultiphysics.HDF5Application",
+        "python_module" : "single_mesh_primal_output_process",
+        "help"          : "",
+        "process_name"  : "",
+        "Parameters" : {
+            "model_part_name" : "Structure",
+            "file_settings" : {
+                "file_access_mode" : "truncate"
+            },
+            "model_part_output_settings" : {
+                "prefix" : "/ModelData"
+            },
+            "nodal_results_settings" : {
+                "list_of_variables": ["DISPLACEMENT", "ROTATION"]
+            }
+        }
+    }
+```
+
+#### Definition of the Adjoint Problem
+In order to define the adjoint problem an additional \*.json-file for the adjoint project parameters is necessary. This input file is in principle very similar to the respective file of the primal analysis. In comparison to a regular file for a linear static analysis three points have to be modified:
+- ```solver_settings``` by using the ```adjoint``` as ```solver_type``` and by the definition of the ```response_function_settings```
+- The input process of the HDF5Application has to be added to the ```list_other_processes``` in order to read the primal solution
+- When defining *Dirichlet* conditions in the ```constraints_process_list``` the ```variable_name``` has to be modified to ```ADJOINT_DISPLACEMENT``` respective ```ADJOINT_ROTATION```
+
+For example the ```solver_settings``` can look like this (Hints for the ```response_function_settings``` are given below):
+
+```python
```

Comment:
This will be outdated one day
Maybe it is better to point to a test for an example (since the tests have to be maintained)

Reply:
Done